### PR TITLE
[CI] Add "defaults" feature for release test configs

### PR DESCRIPTION
**Pull request by bveeramani**
## Why are these changes needed?

The Ray Data release test config contains lots of duplicate configuration (e.g., the working directory and team name). To reduce the maintenance cost and improve readability, this PR adds a feature where you can specify defaults.

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: release/ray_release/config.py [hunk 1]
```diff
@@ -81,11 +81,17 @@ def _test_definition_invariant(
 
 
 def parse_test_definition(test_definitions: List[TestDefinition]) -> List[Test]:
+    defaults = {}
     tests = []
     for test_definition in test_definitions:
+        if "defaults" in test_definition:
+            defaults = test_definition.pop("defaults")
+            continue
+
```
[the remark this answers is not in the document]
Comment (bveeramani) [reply release/ray_release/config.py:90]: Good call. Fixed

File: release/ray_release/config.py [hunk 2]
```diff
@@ -81,11 +81,20 @@ def _test_definition_invariant(
 
 
 def parse_test_definition(test_definitions: List[TestDefinition]) -> List[Test]:
+    default_definition = {}
     tests = []
     for test_definition in test_definitions:
+        if test_definition["name"] == "DEFAULTS":
+            default_definition = test_definition
```
Comment (aslonnie) [inline release/ray_release/config.py:88]: make a deepcopy here? just to be less error prone (and the copy should be cheap)

**Review by aslonnie — APPROVED** [review]

**Comment by aslonnie** [comment]
````
E           AttributeError: 'dict' object has no attribute 'deep_update'
````

please fix the unit test?

**Comment by bveeramani** [comment]
Fixed unit tests and confirmed that release test runs successfully. Going to merge

**Pull request merged**